internals/cli/env_template.go

```diff
@@ -0,0 +1,307 @@
+	"useLine":                  useLine,
+	"hasArgs":                  hasArgs,
+	"argUsages":                argUsages,
+	"flagUsages":               flagUsages,
+	"numFlags":                 numFlags,
+}
+
+// Tmpl executes the given template text on data, writing the result to w.
+func Tmpl(w io.Writer, text string, data interface{}) error {
+	t := template.New("top")
+	t.Funcs(templateFuncs)
+	template.Must(t.Parse(text))
+	return t.Execute(w, data)
+}
+
+// rpad adds padding to the right of a string.
+func rpad(s string, padding int) string {
+	template := fmt.Sprintf("%%-%ds", padding)
+	return fmt.Sprintf(template, s)
+}
+
+func trimRightSpace(s string) string {
+	return strings.TrimRightFunc(s, unicode.IsSpace)
+}
+
+func hasSubCommands(cmd *cobra.Command) bool {
+	return len(operationSubCommands(cmd)) > 0
+}
+
+func hasManagementSubCommands(cmd *cobra.Command) bool {
+	return len(managementSubCommands(cmd)) > 0
+}
+
+func operationSubCommands(cmd *cobra.Command) []*cobra.Command {
+	cmds := []*cobra.Command{}
+	for _, sub := range cmd.Commands() {
+		if sub.IsAvailableCommand() && !sub.HasSubCommands() {
+			cmds = append(cmds, sub)
+		}
+	}
+	return cmds
+}
+
+func managementSubCommands(cmd *cobra.Command) []*cobra.Command {
+	cmds := []*cobra.Command{}
+	for _, sub := range (*cmd).Commands() {
+		if sub.IsAvailableCommand() && sub.HasSubCommands() {
+			cmds = append(cmds, sub)
+		}
+	}
+	return cmds
+}
+
+func decoratedName(cmd cobra.Command) string {
+	return cmd.Name() + " "
+}
+
+func hasArgs(args []Argument) bool {
+	return len(args) > 0
+}
+
+func argUsages(args []Argument) string {
+	buf := new(bytes.Buffer)
+	lines := make([]string, 0, len(args))
+	maxlen := 0
+
+	for _, arg := range args {
+		if arg.Hidden {
+			continue
+		}
+		line := "  "
+		if arg.Placeholder != "" {
+			line += arg.Placeholder
+		} else {
+			line += "<" + arg.Name + ">"
+		}
+
+		line += "\x00"
+		if len(line) > maxlen {
+			maxlen = len(line)
+		}
+
+		line += arg.Description
+		lines = append(lines, line)
+	}
+
+	for _, line := range lines {
+		sidx := strings.Index(line, "\x00")
+		spacing := strings.Repeat(" ", maxlen-sidx)
+		// maxlen + 2 comes from + 1 for the \x00 and + 1 for the (deliberate) off-by-one in maxlen-sidx
+		fmt.Fprintln(buf, line[:sidx], spacing, strings.Replace(line[sidx+1:], "\n", "\n"+strings.Repeat(" ", maxlen+2), -1))
+	}
+
+	return buf.String()
+}
+
+func useLine(c *cobra.Command, args []Argument) string {
+	var useLine string
+
+	if c.HasParent() {
+		useLine = c.Parent().CommandPath() + " " + c.Use
+	} else {
+		useLine = c.Use
+	}
+
+	if c.HasAvailableFlags() && !strings.Contains(useLine, "[flags]") {
+		useLine += " [flags]"
+	}
+
+	for _, arg := range args {
+		if arg.Hidden {
+			continue
+		}
+		if arg.Placeholder != "" {
+			useLine += " " + arg.Placeholder
+		} else {
+			useLine += " <" + arg.Name + ">"
+		}
+	}
+
+	return useLine
+}
+
+func flagUsages(c CommandClause, isGlobal bool) string {
```

Comment:
I'm not sure if this is the piece of code responsible for this, but the help text does not wrap as nicely as it used to.
This is especially visible when looking at flag descriptions:

Before:
![image](https://user-images.githubusercontent.com/13394704/102340673-18959580-3f97-11eb-9f82-671e6eaaa635.png)

After:
![image](https://user-images.githubusercontent.com/13394704/102340587-0156a800-3f97-11eb-9d46-c22f33c747f4.png)

Reply:
Oh, that's an interesting thing. Didn't notice it on the big terminal windows.
Will look into it and adjust it properly. Shouldn't be hard, since we have our own template.

Reply:
Done. Here is the new version (with `secrethub help acl set` as an example)
![Screenshot from 2020-12-21 12-20-23](https://user-images.githubusercontent.com/25607825/102766527-efa24580-4386-11eb-894e-4508e90a9910.png)

Reply:
Awesome! Thanks for the quick fix!